core/src/typecheck/eq.rs

```diff
@@ -105,19 +138,58 @@ impl std::iter::FromIterator<(Ident, (RichTerm, SimpleTermEnvironment))> for Sim
     }
 }
 
-/*
-impl<C: Cache> TermEnvironment for eval::Environment {
-    fn get_then<F, T>(&self, id: &Ident, f: F) -> T
+#[derive(Clone, Copy)]
+pub struct EvalEnvsRef<'a> {
+    pub eval_env: &'a eval::Environment,
+    pub initial_env: &'a eval::Environment,
+}
```

Comment:
It seems the issue here is that we can't construct a `&EvalEnvsOwned` from two `&Environment`s. Do I have that right? Is this a common pattern in rust?

Reply:
Exactly: what I want is not a reference to a tuple but a tuple of references instead. I don't know if it's pattern to define a companion reference type and call it `Ref` like that, but at least it seemed natural.